src/sentry/integrations/bitbucket/repository.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import absolute_import
+
+import six
+
+from uuid import uuid4
+
+from sentry.app import locks
+from sentry.models import OrganizationOption
+from sentry.plugins import providers
+from sentry.models import Integration
+from sentry.utils.http import absolute_uri
+
+from sentry_plugins.exceptions import ApiError
+
+from .webhook import parse_raw_user_email, parse_raw_user_name
+
+
+class BitbucketRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'Bitbucket v2'
+
+    def raise_error(self, error, identity):
+        # TODO(LB): This used to do a lot more. Not sure it's important to handle atm
+        raise error
+
+    def get_client(self, integration_id):
+        from .integration import BitbucketIntegration
+
+        if integration_id is None:
+            raise ValueError('Bitbucket version 2 requires an integration id.')
+
+        model = Integration.objects.get(id=integration_id)
+        return BitbucketIntegration(model).get_client()
+
+    def get_config(self, organization):
+        return [
+            {
+                'name': 'name',
+                'label': 'Repository Name',
+                'type': 'text',
+                'placeholder': 'e.g. getsentry/sentry',
+                'help': 'Enter your repository name, including the owner.',
+                'required': True,
+            }
+        ]
+
+    def validate_config(self, organization, config):
+        """
+        ```
+        if config['foo'] and not config['bar']:
+            raise PluginError('You cannot configure foo with bar')
+        return config
+        ```
+        """
+        if config.get('name'):
+            client = self.get_client(config['integration_id'])
+            try:
+                repo = client.get_repo(config['name'])
+            except Exception as e:
+                self.raise_error(e, identity=client.auth)
```

Comment:
I left this the same, but looking at it, it will fail. What did we want to do with raise_error? It relies on a lot of different stuff from parent classes in the original. Didn't seem super high priority, but I want to check before I change it.